pytket/extensions/qiskit/qiskit_convert.py

```diff
@@ -676,32 +677,58 @@ def _append_if_else_circuit(
     cargs: list[Clbit],
 ) -> None:
     # Get two pytket circuits which implement the true_body and false_body.
+    # Note that else_circ can be None if no false_body is specified.
     if_circ, else_circ = _pytket_circuits_from_ifelseop(
         if_else_op, outer_builder, qargs, cargs
     )
-    # else_circ can be None if no false_body is specified.
-    if isinstance(if_else_op.condition[0], Clbit):
-        if len(bits) != 1:
-            raise NotImplementedError("Conditions on multiple bits not supported")
+
+    if isinstance(if_else_op.condition, (Unary | Binary | Var)):
+        raise NotImplementedError(
+            "Handling of Unary, Var and Binary conditions is not supported yet."
+        )
+
+    if hasattr(if_else_op.condition, "__getitem__") and isinstance(
+        if_else_op.condition[0], Clbit
+    ):
+        condition_bits = [
+            bit
+            for bit in bits
+            if bit.reg_name == if_else_op.condition[0]._register.name  # noqa: SLF001
+            and bit.index[0] == if_else_op.condition[0]._index  # noqa: SLF001
+        ]
+
+        if len(condition_bits) == 0:
+            raise ValueError(
+                "Failed to find any pytket Bit matching Qiskit Clbit in condition for IfElseOp."
+            )
+        if len(condition_bits) > 1:
+            raise ValueError(
+                f"Found condition on {len(condition_bits)},"
+                "only conditions on single individual bits are supported at present."
```

Comment:
I guess we'll only hit this error if `bits` contains duplicates (or multi-indexed bits), so perhaps the error message should say this explicitly.